src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs

```diff
@@ -164,42 +156,21 @@ public bool IsLegacyCSProjPackageReferenceProject
 
                 if (!_isLegacyCSProjPackageReferenceProject.HasValue)
                 {
-                    var restoreProjectStyle = string.Empty;
-
-                    var hasRestoreProjectStyle = _context
-                        .MSBuildProperties
-                        .TryGetValue(ProjectSystemProviderContext.RestoreProjectStyle, out restoreProjectStyle)
-                        && !string.IsNullOrEmpty(restoreProjectStyle);
-
-                    if (!hasRestoreProjectStyle)
+                    // A legacy CSProj can't be CPS, must cast to VSProject4 and *must* have at least one package
+                    // reference already in the CSProj. In the future this logic may change. For now a user must
+                    // hand code their first package reference. Laid out in longhand for readability.
+                    if (AsIVsHierarchy?.IsCapabilityMatch("CPS") ?? true)
```

Comment:
This is not a new check, it's always been there. Currently we skip making CPS based projects to be legacy csproj project system and also currently CPS based projects doesn't support VSProject4 so it will never fall here.

ServiceFabric n other CPS are already using Packages.config and will continue to do so.